lib/replicache.dart

```diff
@@ -135,40 +139,49 @@ class Replicache implements ReadTransaction {
   String get remote => _remote;
   String get clientViewAuth => _clientViewAuth;
 
-  /// Puts a single value into the database in its own transaction.
-  Future<void> _put(String key, dynamic value) async {
+  Future<void> _put(int transactionId, String key, dynamic value) async {
     await _opened;
-    return _result(await _checkChange(
-        await _invoke(_name, 'put', {'key': key, 'value': value})));
+    await _invoke(_name, 'put', {
+      'transactionId': transactionId,
+      'key': key,
+      'value': value,
+    });
   }
 
   Future<dynamic> _get(int transactionId, String key) async {
     await _opened;
-    return _result(await _invoke(_name, 'get', {
+    final result = await _invoke(_name, 'get', {
       'transactionId': transactionId,
       'key': key,
-    }));
+    });
+    if (!result['has']) {
+      return null;
+    }
+    return result['value'];
   }
 
   /// Get a single value from the database.
   Future<dynamic> get(String key) => query((tx) => tx.get(key));
 
   Future<bool> _has(int transactionId, String key) async {
     await _opened;
-    return _result(await _invoke(_name, 'has', {
+    final result = await _invoke(_name, 'has', {
```

Comment:
Seriously. I started playing with this too. We can mock the MethodChannel but that seems like it would not catch errors like these.

I also started looking at cgo to see if we can just use ffi (at least in tests)

Reply:
See the bug I created for alternate ideas. It might be best to just get flutter/desktop working.